transports/quic/src/endpoint.rs

```diff
@@ -142,6 +145,23 @@ impl Endpoint {
 
         Ok(endpoint)
     }
+
+    pub fn socket_addr(&self) -> &SocketAddr {
+        &self.socket_addr
+    }
+
+    pub fn try_send(
+        &mut self,
+        to_endpoint: ToEndpoint,
+        cx: &mut Context<'_>,
+    ) -> Result<Result<(), ToEndpoint>, SendError> {
+        match self.to_endpoint.poll_ready_unpin(cx) {
+            Poll::Ready(Ok(())) => {}
+            Poll::Ready(Err(err)) => return Err(err),
+            Poll::Pending => return Ok(Err(to_endpoint)),
+        };
+        self.to_endpoint.start_send(to_endpoint).map(Ok)
```

Comment:
Have you consider just using [`Sender::try_send`](https://docs.rs/futures/latest/futures/channel/mpsc/struct.Sender.html#method.try_send) instead of the poll-based functions here?

Reply:
[`Sender::try_send`](https://docs.rs/futures/latest/futures/channel/mpsc/struct.Sender.html#method.try_send) does not take a `Context` and thus does not wake up the underlying task once it can make progress.

Reply:
Yes I did, but we need the poll-based functions so that the waker is registered before we return `Poll::Pending` in `Connection::poll_event`.

Reply:
Makes sense, can we document this in the function docs?

Reply:
Done in 4c3229b.